framework/rest/ActiveDataFilter.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\rest;
+
+/**
+ * ActiveDataFilter allows composition of the filter condition in format suitable for [[\yii\db\QueryInterface::where()]].
+ *
+ * @see DataFilter
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.0.10
+ */
+class ActiveDataFilter extends DataFilter
+{
+    /**
+     * @var array map of filter condition keywords to build methods.
+     * These methods are used by [[buildCondition]] to build the actual filter conditions.
+     */
+    public $conditionBuilders = [
+        '$and' => 'buildConjunctionCondition',
```

Comment:
is there an RFC for this or similar filtering syntax? yii2 rest follows json hypermedia, hateoas and hal natively very closely so i think we should start searching there for filtering implementation.